Create Bishop moves  for light pieces

Mrsatatima opened the pull request:
A function for Bishop moves, it generates all possible for Bishop pieces
while making sure pieces doesn't go off board and jump over pieces
it captures enemy pieces (dark) and stop at that square.

Mrsatatima committed:
Create Bishop moves  for light pieces

A function for Bishop moves, it generates all possible for Bishop pieces
while making sure pieces doesn't go off board and jump over pieces
it captures enemy pieces (dark) and stop at that square.

Bilaerl commented on engine.py:
```diff
@@ -85,10 +85,25 @@ def get_bishop_moves(self, r, c, moves):
+		if self.light_to_move:
+			direction = [(-1,-1), (1,1),(1,-1), (-1,1)] # possible direction
+			for d in direction:
+				for i in range(1,8):
```
Use something that will scale to new data, using range(1,8) will only work if board size is 8 by 8, if for some reason someone changes the board size your code won't scale. Instead try;

for i in range(len(self.board)):

Bilaerl commented on engine.py:
```diff
@@ -85,10 +85,25 @@ def get_bishop_moves(self, r, c, moves):
+				for i in range(1,8):
+					rownum = r + d[0] * i
+					colnum = c + d[1] * i
+					if 0 <= rownum <8 and 0 <= colnum < 8: # making sure r and c on board
```
Use parenthesis to isolate different parts of a boolean expression

if (0 <= rownum < 8) and (0 <= colnum < 8)

And use len(self.board) instead of 8 for scalability

Bilaerl commented on engine.py:
```diff
@@ -85,10 +85,25 @@ def get_bishop_moves(self, r, c, moves):
+					if 0 <= rownum <8 and 0 <= colnum < 8: # making sure r and c on board
+						if self.board[rownum][colnum] == "  ":
+							moves.append(Move((r,c), (rownum,colnum), (self.board)))
+						elif self.board[rownum][colnum][1] == "d": # capture enemy piece
```
change comment to "if square has opponent piece"

Bilaerl commented on engine.py:
```diff
@@ -85,10 +85,25 @@ def get_bishop_moves(self, r, c, moves):
+					rownum = r + d[0] * i
+					colnum = c + d[1] * i
+					if 0 <= rownum <8 and 0 <= colnum < 8: # making sure r and c on board
+						if self.board[rownum][colnum] == "  ":
```
Add comment "if square is empty"

Bilaerl commented on engine.py:
```diff
@@ -85,10 +85,25 @@ def get_bishop_moves(self, r, c, moves):
- 		pass
+		##TODO
+		if self.light_to_move:
+			direction = [(-1,-1), (1,1),(1,-1), (-1,1)] # possible direction
```
Change _direction_ from list of tuples to a tuple of tuples, since you don't need to edit it in runtime. And make spacing between elements uniform. i.e 
((-1,-1),(1,1),(1,-1),(-1,1))  : no spacing
or
((-1,-1), (1,1), (1,-1), (-1,1)) : single space

Bilaerl reviewed (changes requested):
Nice algorithm, I'm impressed.
Appropriate commit message.
Keep it up

drak00 committed:
FIX ZeroDivisionError in animate

Mrsatatima committed:
Merge branch 'light_team' of github.com:drak00/Chess into light_team

Mrsatatima committed:
Create Bishop moves  for Dark pieces

A function for Bishop moves, it generates all possible for Bishop pieces
while making sure pieces doesn't go off board and jump over pieces
it captures enemy pieces (light) and stop at that square.

Mrsatatima reviewed (commented):
okay

Mrsatatima commented:
will make changes
